app/models/schedule.rb

```diff
@@ -159,10 +163,13 @@ def timeline
     end 
   end
 
-end
-
-
-
-
-
+private
+  def assign_root
+    if root_task.nil?
+      #find first defined task and set it equal to root
+      self.root_task = Task.where(schedule_id: self.id).first
+      self.save
```

Comment:
This results in a SystemStackError. You're triggering #assign_root after #save, and then triggering #save within this method. So... :boom: